Add support to import AST via Standard JSON. (#14171)
ethereum/solidity: docs/using-the-compiler.rst

            // If files are used, their directories should be added to the command line via
            // `--allow-paths <path>`.
          ]

>           // If language was set to "SolidityAST", the ast need to be supplied under the "ast" key.
>           // However, right now the "SolidityAST" language is just an experimental feature:
>           // - it is not guaranteed that it will generate valid results on invalid ASTs.
>           // - it also re-performs analysis: so it might be possible that not
>           //   all information defined by the provided AST will be used, because it may recompute some AST elements.
>           "ast": { ... } //  the format need to be the same as generated with `solc --ast-compact-json`.

Reviewer: ```suggestion
          // If language is set to "SolidityAST", an ast needs to be supplied under the "ast" key.
          // Note that importing ASTs is experimental and in particular that:
          // - importing invalid ASTs can produce undefined results and
          // - no proper error reporting is available on invalid ASTs.
          // Furthermore, note that the AST import only consumes the fields of the AST as
          // produced by the compiler in "stopAfter": "parsing" mode and then re-performs
          // analysis, so any analysis-based annotations of the AST are ignored upon import.
          "ast": { ... } // formatted as the json ast requested with the ``ast`` output selection.
```

            // If files are used, their directories should be added to the command line via
            // `--allow-paths <path>`.
          ]
          // If language is set to "SolidityAST", an AST needs to be supplied under the "ast" key.
          // Note that importing ASTs is experimental and in particular that:
          // - importing invalid ASTs can produce undefined results and
          // - no proper error reporting is available on invalid ASTs.
          // Furthermore, note that the AST import only consumes the fields of the AST as
          // produced by the compiler in "stopAfter": "parsing" mode and then re-performs
          // analysis, so any analysis-based annotations of the AST are ignored upon import.

>           "ast": { ... } // formatted as the produced of the ``ast`` output selection.

Reviewer: ```suggestion
          "ast": { ... } // formatted as the json ast requested with the ``ast`` output selection.
```
Sorry, my earlier suggestion was also weird garbage here :-).
